backend/open_webui/utils/middleware.py

```diff
@@ -937,11 +938,20 @@ async def process_chat_payload(request, form_data, user, metadata, model):
                         or source.get("source", {}).get("id", None)
                         or "N/A"
                     )
-                    if citation_id not in citation_idx:
-                        citation_idx[citation_id] = len(citation_idx) + 1
+                    if citation_id not in citation_metadata:
+                        source_counter += 1
+                        citation_metadata[citation_id] = {
+                            "file_id": doc_meta.get("file_id", None),
+                            "source_id": source_counter
+                        }
+                    
+                    file_id = citation_metadata[citation_id]["file_id"]
+                    source_id = citation_metadata[citation_id]["source_id"]
+                    
                     context_string += (
-                        f'<source id="{citation_idx[citation_id]}"'
+                        f'<source id="{source_id}"'
                         + (f' name="{source_name}"' if source_name else "")
+                        + (f' file_id="{file_id}"' if file_id else "")
```

Comment:
If this becomes
```python
f' file_id="{doc_meta.get("file_id", None)}"' ...
```
can we remove the rest of the changes?

Just want to keep diff with upstream as small as possible.